src/Analyser/MutatingScope.php

```diff
@@ -467,10 +466,6 @@ public function afterOpenSslCall(string $openSslFunctionName): self
 	/** @api */
 	public function hasVariableType(string $variableName): TrinaryLogic
 	{
-		if ($this->isGlobalVariable($variableName)) {
-			return TrinaryLogic::createYes();
```

Comment:
I guess that the same thing can be done only by bring these conditions inside !isset, before canAnyVariableExist

Reply:
what do you mean? my problem here was that scope merging does not work properly because the expression was missing. `hasVariableType()` always answered correctly I think

Reply:
Sorry, I read your comments in pull request afterwards never mind with this